consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/jsonrpc/methods/IbftGetValidatorsByBlockNumber.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.consensus.ibft.jsonrpc.methods;
+
+import tech.pegasys.pantheon.consensus.ibft.IbftBlockInterface;
+import tech.pegasys.pantheon.ethereum.chain.Blockchain;
+import tech.pegasys.pantheon.ethereum.core.BlockHeader;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.JsonRpcRequest;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.methods.AbstractBlockParameterMethod;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.methods.JsonRpcMethod;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.parameters.BlockParameter;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.parameters.JsonRpcParameter;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.queries.BlockchainQueries;
+
+import java.util.Optional;
+
+public class IbftGetValidatorsByBlockNumber extends AbstractBlockParameterMethod
+    implements JsonRpcMethod {
+
+  private final Blockchain blockchain;
+  private final IbftBlockInterface ibftBlockInterface;
+
+  public IbftGetValidatorsByBlockNumber(
+      final BlockchainQueries blockchainQueries,
```

Comment:
Having both BlockChainQueries AND Blockchain is a smell - i.e. I'm assuming the queries should hide the underlying blockchain.

I _suspect_ the smell is that AbstractBlockParameterMethod should actually be 2 things:
1. AbstractJsonParameterHandler
2.BlockchainQuerier

That way, this class can re-use the AbstractJsonParameterHandler to do the json bits, but then wouldn't need both a BlockchainQuery AND Blockchain...